src/wagtail_ai/views.py

```diff
@@ -98,14 +106,70 @@ def process(request) -> JsonResponse:
     try:
         response = handler(prompt=prompt, text=prompt_form.cleaned_data["text"])
     except AIHandlerException as e:
-        return JsonResponse({"error": str(e)}, status=400)
+        return ErrorJsonResponse(str(e), status=400)
     except Exception:
         logger.exception("An unexpected error occurred.")
-        return JsonResponse({"error": _("An unexpected error occurred.")}, status=500)
+        return ErrorJsonResponse(_("An unexpected error occurred."))
 
     return JsonResponse({"message": response})
 
 
+def user_has_permission_for_image(user, image):
+    from wagtail.images.permissions import permission_policy
+
+    return permission_policy.user_has_permission_for_instance(user, "choose", image)
+
+
+def describe_image(request) -> JsonResponse:
```

Comment:
**Suggestion:** We could an audit trail for using the AI functionality. https://docs.wagtail.org/en/stable/extending/audit_log.html

Reply:
Raised #85.